dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -194,6 +199,52 @@ private static Validator createValidator() {
         return validatorFactory.getValidator();
     }
 
+    /**
+     * Decide whether a gitReference is excluded, given a workflow/service's filters
+     * @param gitRefPath Path.of(gitReference) for glob matching with PathMatcher
+     * @param filters Filters specified for a workflow/service in .dockstore.yml
+     * @return
+     */
+    public static boolean filterGitReference(final Path gitRefPath, final Filters filters) {
+        final List<String> branches = filters.getBranches();
+        final List<String> tags = filters.getTags();
+
+        // If no filters specified, accept anything
+        if (branches.isEmpty() && tags.isEmpty()) {
+            return true;
+        }
+
+        List<String> patterns;
+        if (gitRefPath.startsWith("refs/heads/")) {
+            patterns = branches;
+        } else if (gitRefPath.startsWith("refs/tags/")) {
+            patterns = tags;
+        } else {
+            // Reaching this is unexpected - should have a warning here.
```

Comment:
Yes, it should at least do a `LOG.warn` probably error
In fact, this seems so unexpected it should probably throw an exception and die hard rather than (only) log